internal/hooks/tracking.go

```diff
@@ -115,3 +115,17 @@ func removeFromCommaSeparatedList(list string, items ...string) string {
 	set.Delete(items...)
 	return strings.Join(set.List(), ",")
 }
+
+func AddHookToPendingAnnotation(obj client.Object, hooks ...runtimecatalog.Hook) {
```

Comment:
Can we also use this func in MarkAsDone? just to deduplicate code and get some test coverage without actually adding new tests :)